db/migrate/20170824202628_cleanup_root_guids_from_reshares.rb

```diff
@@ -0,0 +1,13 @@
+class CleanupRootGuidsFromReshares < ActiveRecord::Migration[5.1]
+  def up
+    # rubocop:disable Rails/SkipsModelValidations
+    Reshare.joins("LEFT OUTER JOIN posts as root ON root.guid = posts.root_guid")
+           .where("root.id is NULL AND posts.root_guid is NOT NULL")
+           .update_all(root_guid: nil)
+    # rubocop:enable Rails/SkipsModelValidations
+  end
+
+  def down
+    raise ActiveRecord::IrreversibleMigration
```

Comment:
Do we really need this `IrreversibleMigration` here? This will block migrations from roll back. But it feels safe to roll back here by doing nothing: in this case we will stay with `root_guid=nil` for deleted roots which is fine for pre-migration state as well. So why fail?

Reply:
You are right, I removed it, thanks 👍